autoblocks/_impl/configs/config.py

```diff
@@ -0,0 +1,246 @@
+import abc
+import asyncio
+import json
+import logging
+from datetime import timedelta
+from typing import Any
+from typing import Callable
+from typing import Dict
+from typing import Generic
+from typing import Optional
+from typing import TypeVar
+
+from autoblocks._impl import global_state
+from autoblocks._impl.config.constants import API_ENDPOINT
+from autoblocks._impl.configs.models import DangerouslyUseUndeployedRemoteConfig
+from autoblocks._impl.configs.models import DangerouslyUseUndeployedWithRevision
+from autoblocks._impl.configs.models import LatestDangerouslyUseUndeployed
+from autoblocks._impl.configs.models import LatestRemoteConfig
+from autoblocks._impl.configs.models import RemoteConfig
+from autoblocks._impl.configs.models import RemoteConfigResponse
+from autoblocks._impl.configs.models import RemoteConfigWithVersion
+from autoblocks._impl.util import AnyTask
+from autoblocks._impl.util import AutoblocksEnvVar
+from autoblocks._impl.util import get_running_loop
+
+log = logging.getLogger(__name__)
+
+AutoblocksConfigValueType = TypeVar("AutoblocksConfigValueType")
+
+
+def is_testing_context() -> bool:
+    """
+    Note that we check for the presence of the CLI environment
+    variable and not the test case contextvars because the
+    contextvars aren't set until run_test_suite is called,
+    whereas a prompt manager might have already been imported
+    and initialized by the time run_test_suite is called.
+    """
+    return bool(AutoblocksEnvVar.CLI_SERVER_ADDRESS.get())
+
+
+def config_revisions_map() -> dict[str, str]:
+    """
+    The AUTOBLOCKS_CONFIG_REVISIONS environment variable is a JSON-stringified
+    map of config IDs to revision IDs. This is set in CI test runs triggered
+    from the UI.
+    """
+    if not is_testing_context():
+        return {}
+
+    config_revisions_raw = AutoblocksEnvVar.CONFIG_REVISIONS.get()
+    if not config_revisions_raw:
+        return {}
+    return json.loads(config_revisions_raw)  # type: ignore
+
+
+def make_request_url(config: RemoteConfig) -> str:
+    config_id = config.id
+    base = f"{API_ENDPOINT}/configs/{config_id}"
+    revision_id = config_revisions_map().get(config_id)
+    if revision_id and is_testing_context():
+        return f"{base}/revisions/{revision_id}"
```

Comment:
super nit:

```python
if is_testing_context() and revision_id := config_revisions_map().get(config_id):
    return f"{base}/revisions/{revision_id}"
```

just so you don't even hit the `config_revisions_map().get(config_id)` code unless you're in a testing context